Remove `rust=rustc` alias

Ericson2314 opened the pull request:
I was asked to add this for backwards comparability when I renamed `rust`
to `rustc` and added `rustcMaster`. It has been a few months so I'd hope
deprecating this is acceptable.

Also racer should have Rust's source as an explicit dependency because
compile time dependencies are not propagated by default.

jagajaga commented on pkgs/development/tools/rust/racer/default.nix:
```diff
@@ -1,4 +1,4 @@
-{stdenv, fetchgit, rust, makeWrapper }:
+{stdenv, fetchgit, rustc, makeWrapper }:
 
 let
```
Error, isn't it?

jagajaga commented on pkgs/development/tools/rust/racer/default.nix:
```diff
@@ -1,4 +1,4 @@
+{stdenv, fetchgit, rustc, makeWrapper }:
 
 let
   rustSrc = rust.src;
```
Error, isn't it? Should be rustc.src if you changed argument like this.

Ericson2314 replied:
Oh wow. Stupid mistake. Sorry about that.

Ericson2314 replied:
I'd grep'd and thought I got everything.

jagajaga commented:
Update and force push. 
Oh, please, can you change racer's version to 2015-01-07 too?

Ericson2314 commented:
I can fix the version. Was racer being developed against 0.12? No that 1.0-alpha is out I assume we will want to drop 0.12...

jagajaga commented:
I'll fix it by myself.

Ericson2314 commented:
Ah, hadn't the override in Rust's `callPackage` invocation. I have tested that racer evaluates to a valid derivation this time.

jagajaga commented on pkgs/development/tools/rust/racer/default.nix:
```diff
@@ -13,16 +9,16 @@ stdenv.mkDerivation rec {
   };
 
-  buildInputs = [ rust makeWrapper ];
+  buildInputs = [ rustc rustc.src makeWrapper ];
```
Why are you adding src as an additional dep?

Ericson2314 commented:
So I thought after doing some research and talking to people on IRC that it wouldn't be registered as a dep otherwise, because compile time steps are not propagated my default.

Ericson2314 commented:
But going back I hear that perhaps any subexpression that evaluates to a derivation triggers a dependency.

jagajaga commented:
Indeed it's a useless one, because `src` is not a `buildInput` itself.
Please remove it.

Ericson2314 committed:
Remove `rust=rustc` alias

I was asked to add this for backwards comparability when I renamed `rust`
to `rustc` and added `rustcMaster`. It has been a few months so I'd hope
deprecating this is acceptable.

Ericson2314 commented:
Thank you!
